src/utils/components/SelectSnapshot/useSnapshots.ts

```diff
@@ -0,0 +1,70 @@
+import { useMemo } from 'react';
+
+import {
+  modelToGroupVersionKind,
+  ProjectModel,
+  VolumeSnapshotModel,
+} from '@kubevirt-ui/kubevirt-api/console';
+import { V1VolumeSnapshotStatus } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { K8sResourceCommon, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+type UseSnapshotsReturnType = {
+  error: Error;
+  projectsLoaded: boolean;
+  projectsNames: string[];
+  snapshots: VolumeSnapshotKind[];
+  snapshotsLoaded: boolean;
+};
+
+export type VolumeSnapshotKind = K8sResourceCommon & {
```

Comment:
if u are exporting a type better to put in a new tyhpes.ts near util, other components can consume it not from a hook file, just make it more organize